exchange/exchange.go

```diff
@@ -865,6 +865,59 @@ func (e *exchange) getBidCacheInfo(bid *pbsOrtbBid, auction *auction) (cacheInfo
 	return
 }
 
+// getAuctionCurrencyRates returns the constant currency conversion rates that will be used to process this
+// auction. Parameter customRates contains, if any, the bidRequest custom currency conversion rates
+// defined in bidRequest.Ext.  If usePbsRates flag is set to false, all previous Conversions values are discarded
+// and this function merely substitutes Conversions with customRates. But if usePbsRates was ommitted or set to
+// true, customRates' values update or add to the Conversions map
+func (e *exchange) getAuctionCurrencyRates(customRates *openrtb_ext.ExtRequestCurrency) currency.Conversions {
+	pbsConversions := e.currencyConverter.Rates()
```

Comment:
Any reason why we are doing the default conversions first and then discarding them if custom conversion rates are present? I think it would make more sense and be more efficient to apply the custom conversion rates first if they exist and fallback to default only if custom conversion rates don't exist or don't apply

Reply:
Good point. Also refactored the code a bit and provided a couple more test cases.